extensions/amp-experiment/1.0/amp-experiment.js

```diff
@@ -34,31 +39,50 @@ export class AmpExperiment extends AMP.BaseElement {
 
   /** @override */
   buildCallback() {
-    return getServicePromiseForDoc(this.getAmpDoc(), 'variant')
-        .then(variantsService => {
-          try {
-            const config = this.getConfig_();
-            const results = Object.create(null);
-            const variants = Object.keys(config).map(experimentName => {
-              return allocateVariant(
-                  this.getAmpDoc(), experimentName, config[experimentName])
-                  .then(variantName => {
-                    results[experimentName] = variantName;
-                  });
-            });
-
-            /** @private @const {!Promise<!Object<string, ?string>>} */
-            const experimentVariants = Promise.all(variants)
-                .then(() => results)
-                .then(this.applyExperimentVariants_.bind(this, config));
-
-            variantsService.init(experimentVariants);
-          } catch (e) {
-            // Ensure downstream consumers don't wait for the promise forever.
-            variantsService.init(Promise.resolve({}));
-            throw e;
-          }
+
+    const buildCallbackPromises = [
+      getServicePromiseForDoc(this.getAmpDoc(), 'variant'),
+      this.isExperimentEnabled_(),
+    ];
+
+    return Promise.all(buildCallbackPromises).then(responses => {
+      const variantsService = responses[0];
+      const enabled = responses[1];
+
+      if (!enabled) {
+        user().error(TAG, 'Experiment amp-experiment-1.0 is not enabled.');
+
+        // Ensure downstream consumers don't wait for the promise forever.
+        variantsService.init(Promise.resolve({}));
+
+        return Promise.reject(
```

Comment:
Is `Promise.reject()` handled well by `buildCallback()`. It won't lead to uncaught dev error right

Reply:
In the screenshots, the only ugly error I saw was "Uncaught Error: _reported_ ...". Which I found to also happen for other experiments (reCAPTCHA). Which I think is a recent thing in master, as it originally did not do that.

That being said, this code is essentially copy-paste from the reCAPTCHA experiment, so I assume it does not? 🤔 

I'll wait for your reply before merging @zhouyx 😄

Reply:
I looked at the source code. 
https://github.com/ampproject/amphtml/blob/47bd44948d0c57bd3c8e8b9cbb19b92c1e7c3b7a/src/custom-element.js#L536-L542

We reportError on this, which I believe should not be reported. But that's not blocking, we can fix it later.

Reply:
Want me to create an issue for this? 😄 But glad this is not blocking the PR 👍